force-app/main/default/classes/ALLO_AllocationsRecalculateService.cls

```diff
@@ -168,7 +168,11 @@ public inherited sharing class ALLO_AllocationsRecalculateService {
         // Allocations should not be copied to the target if the target object's amount is null
         if (amount == null) {
             return;
-        }
+        } else if (sourceAmount == 0 && convertToPercentages) {
+            // Allocations should not be copied to target if source amount is zero and convert to percentage is true
+                return;
+            }
```

Comment:
Minor and non-blocking, but the indentation is off here.

Reply:
Ha I saw that too, OCD kicked in 😅